vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java

```diff
@@ -0,0 +1,617 @@
+        @Override public void completed() { }
+        @Override public void failed(Throwable t) {
+            log.log(WARNING, "Exception writing response data", t);
+        }
+    };
+
+    private static final ContentChannel ignoredContent = new ContentChannel() {
+        @Override public void write(ByteBuffer buf, CompletionHandler handler) { handler.completed(); }
+        @Override public void close(CompletionHandler handler) { handler.completed(); }
+    };
+
+    private static final String CREATE = "create";
+    private static final String CONDITION = "condition";
+    private static final String ROUTE = "route"; // TODO jonmv: set for everything except Get
+    private static final String FIELD_SET = "fieldSet";
+    private static final String SELECTION = "selection";
+    private static final String CLUSTER = "cluster"; // TODO jonmv: set for Get
+    private static final String CONTINUATION = "continuation";
+    private static final String WANTED_DOCUMENT_COUNT = "wantedDocumentCount";
+    private static final String CONCURRENCY = "concurrency";
+    private static final String BUCKET_SPACE = "bucketSpace";
+
+    private final Clock clock;
+    private final Metric metric; // TODO jonmv: make response class which logs on completion/error
+    private final DocumentApiMetrics metrics;
+    private final DocumentOperationExecutor executor;
+    private final DocumentOperationParser parser;
+    private final Map<String, Map<Method, Handler>> handlers;
+
+    @Inject
+    public DocumentV1ApiHandler(Clock clock,
+                                Metric metric,
+                                MetricReceiver metricReceiver,
+                                DocumentmanagerConfig documentManagerConfig,
+                                ClusterListConfig clusterListConfig,
+                                AllClustersBucketSpacesConfig bucketSpacesConfig) {
+        this(clock,
+             new DocumentOperationExecutor(documentManagerConfig, clusterListConfig, bucketSpacesConfig, clock),
+             new DocumentOperationParser(documentManagerConfig),
+             metric,
+             metricReceiver);
+    }
+
+    DocumentV1ApiHandler(Clock clock, DocumentOperationExecutor executor, DocumentOperationParser parser,
+                         Metric metric, MetricReceiver metricReceiver) {
+        this.clock = clock;
+        this.executor = executor;
+        this.parser = parser;
+        this.metric = metric;
+        this.metrics = new DocumentApiMetrics(metricReceiver, "documentV1");
+        this.handlers = defineApi();
+    }
+
+    @Override
+    public ContentChannel handleRequest(Request rawRequest, ResponseHandler rawResponseHandler) {
+        metric.add("handled.requests", 1, contextFor(rawRequest));
+        ResponseHandler responseHandler = new ResponseHandler() {
+            @Override public ContentChannel handleResponse(Response response) {
+                return null;
+            }
+        };
```

Comment:
Oops, leftover from wrapping response handler in another response handler to report metrics ...